vendors/infineon/secure_elements/pkcs11/iot_pkcs11_trustx.c

```diff
@@ -0,0 +1,3122 @@
+
+    /* Clean up memory if there was an error parsing the template. */
+    if( xResult != CKR_OK )
+    {
+        if( pxFindObjectLabel != NULL )
+        {
+            vPortFree( pxFindObjectLabel );
+            pxSession->pxFindObjectLabel = NULL;
+        }
+    }
+
+    return xResult;
+}
+
+/**
+ * @brief Query the objects of the requested type.
+ */
+CK_DEFINE_FUNCTION( CK_RV, C_FindObjects )( CK_SESSION_HANDLE xSession,
+                                            CK_OBJECT_HANDLE_PTR pxObject,
+                                            CK_ULONG ulMaxObjectCount,
+                                            CK_ULONG_PTR pulObjectCount )
+{   /*lint !e9072 It's OK to have different parameter name. */
+	CK_RV xResult = PKCS11_SESSION_VALID_AND_MODULE_INITIALIZED(xSession);
+    BaseType_t xDone = pdFALSE;
+    P11SessionPtr_t pxSession = prvSessionPointerFromHandle( xSession );
+
+    CK_BYTE_PTR pcObjectValue = NULL;
+    uint32_t xObjectLength = 0;
+    CK_BBOOL xIsPrivate = CK_TRUE;
+    CK_BYTE xByte = 0;
+    CK_OBJECT_HANDLE xPalHandle = CK_INVALID_HANDLE;
+    uint32_t ulIndex;
+
+    /*
+     * Check parameters.
+     */
+    if( ( CKR_OK != xResult ) ||
+    	( NULL == pxObject ) ||
+        ( NULL == pulObjectCount ) )
+    {
+        xResult = CKR_ARGUMENTS_BAD;
+        xDone = pdTRUE;
+    }
+
+    if( pxSession->pxFindObjectLabel == NULL )
+    {
+        xResult = CKR_OPERATION_NOT_INITIALIZED;
+        xDone = pdTRUE;
+    }
+
+    if( 0u == ulMaxObjectCount )
+    {
+        xResult = CKR_ARGUMENTS_BAD;
+        xDone = pdTRUE;
+    }
+
+    if( 1u != ulMaxObjectCount )
+    {
+        PKCS11_WARNING_PRINT( ( "WARN: Searching for more than 1 object not supported. \r\n" ) );
+    }
+
+    if( ( pdFALSE == xDone ) && ( ( CK_BBOOL ) CK_TRUE == pxSession->xFindObjectComplete ) )
+    {
+        *pulObjectCount = 0;
+        xResult = CKR_OK;
+        xDone = pdTRUE;
+    }
+
+    /* TODO: Re-inspect this previous logic. */
+    if( ( pdFALSE == xDone ) )
+    {
+        /* Try to find the object in module's list first. */
+        prvFindObjectInListByLabel( pxSession->pxFindObjectLabel, strlen( ( const char * ) pxSession->pxFindObjectLabel ), &xPalHandle, pxObject );
+
+        /* Check with the PAL if the object was previously stored. */
+        if( *pxObject == CK_INVALID_HANDLE )
+        {
+            xPalHandle = PKCS11_PAL_FindObject( pxSession->pxFindObjectLabel, ( uint8_t ) strlen( ( const char * ) pxSession->pxFindObjectLabel ) );
+        }
+
+        if( xPalHandle != CK_INVALID_HANDLE && xPalHandle != eAwsDevicePrivateKey)
+        {
+            xResult = PKCS11_PAL_GetObjectValue( xPalHandle, &pcObjectValue, &xObjectLength, &xIsPrivate );
+
+            if( xResult == CKR_OK )
+            {
+                for( ulIndex = 0; ulIndex < xObjectLength; ulIndex++ )
+                {
+                    xByte |= pcObjectValue[ ulIndex ];
+                }
+
+                if( xByte == 0 ) /* Deleted o
```

Comment:
Is there a #define that represents that max size of an ECDSA point pair (encoded or otherwise) for this implementation? Could that be used here instead?

Reply:
reworked